test/e2e/network_test.go

```diff
@@ -148,6 +153,32 @@ func testEastWestNetworking(t *testing.T) {
 	}
 }
 
+// collectDeploymentLogs collects logs of multiple deployments
+// through a call to writePodLogs with a pod labelSelector
+func (tc *testContext) collectDeploymentLogs(deployments ...*appsv1.Deployment) {
```

Comment:
Is there a reason for taking deployments as an input when all call sites are passing in just one?

This function could figure out the pods in the deployment and make individual calls to for collecting and writing out the pod logs.

Reply:
I understood it as deployment calls have multiple pods associated with them and we want to read logs from each pod

Reply:
I think one of the suggestions here is to change this function's signature to only take in 1 parameter `(deployment *appsv1.Deployment)` instead of making it variadic. I second this approach.

Reply:
> I understood it as deployment calls have multiple pods associated with them and we want to read logs from each pod

What you are doing is taking a list of deployments and traversing them

Reply:
Ok I see my mistake now - I was traversing a list of deployments and collecting logs associated with the first pod of each deployment. I will change this function to take in one deployment and then traverse a list of pods associated with that one deployment to collect logs for each pod individually.